test-e2e/utils/steps_file.ts

```diff
@@ -239,6 +239,48 @@ const stepsObj = {
   writeClipboard: async function (this: CodeceptJS.I, text: string) {
     await this.executeScript((text) => navigator.clipboard.writeText(text), text);
   },
+  scrollToShelf: async function (this: CodeceptJS.I, playlistId: string) {
+    const targetSelector = `[data-mediaid="${playlistId}"]`;
+
+    let tries = 5;
+    let visible = 0;
+
+    do {
+      visible = await this.grabNumberOfVisibleElements(targetSelector);
+
+      if (visible === 0) {
+        this.scrollTo('[role="row"]:last-child');
+        this.wait(0.2);
+      }
+    } while (visible === 0 && tries--);
+
+    assert.strictEqual(visible, 1, `Shelf row not found with id '${playlistId}'`);
+
+    this.scrollTo(targetSelector);
+  },
+  mockTimeAs: async function (this: CodeceptJS.I, time: string, locale: string = 'nl-NL') {
+    return this.usePlaywrightTo(`Mock current time as ${time}`, async ({ page }) => {
+      const currentDate = new Date().toLocaleDateString(locale);
+      const mockedNow = new Date(`${currentDate} ${time}`).valueOf();
```

Comment:
Use UTC times instead of a locale. You will also need to update the mocked times to match the schedule UTC time.

```suggestion
  mockTimeAs: async function (this: CodeceptJS.I, hours: number, minutes: number, seconds: number) {
    return this.usePlaywrightTo(`Mock current time as ${time}`, async ({ page }) => {
      const today = new Date();
      today.setUTCHours(hours, minutes, seconds, 0);
      const mockedNow = today.toJSON();
```